plugins/occt/module/vtkF3DOCCTReader.cxx

```diff
@@ -651,11 +653,30 @@ int vtkF3DOCCTReader::RequestData(
     STEPCAFControl_Reader reader;
     TransferToDocument(this, reader, doc);
   }
-  else // FILE_FORMAT::IGES
+  else if (this->FileFormat == FILE_FORMAT::IGES)
   {
     IGESCAFControl_Reader reader;
     TransferToDocument(this, reader, doc);
   }
+  else if (this->FileFormat == FILE_FORMAT::XBF)
+  {
+    try
+    {
+      Handle(TDocStd_Application) app = new TDocStd_Application();
+      BinXCAFDrivers::DefineFormat(app);
+      if (app->Open(this->GetFileName().c_str(), doc) != PCDM_RS_OK)
+      {
+        vtkErrorWithObjectMacro(this, "Failed to read XBF file");
```

Comment:
is this reachable with an invalid file ?

Reply:
Haven't looked into it yet but this one should be